monkey/monkey_island/cc/repository/mongo_credentials_repository.py

```diff
@@ -0,0 +1,80 @@
+from typing import Sequence
+
+from common.credentials import Credentials
+from monkey_island.cc.database import mongo
+from monkey_island.cc.repository import RemovalError, RetrievalError, StorageError
+from monkey_island.cc.repository.i_credentials_repository import ICredentialsRepository
+
+
+class MongoCredentialsRepository(ICredentialsRepository):
+    """
+    Store credentials in a mongo database that can be used to propagate around the network.
+    """
+
+    def get_configured_credentials(self) -> Sequence[Credentials]:
+        try:
+            configured_credentials = []
+            list_configured_credentials = list(mongo.db.configured_credentials.find({}))
+            for c in list_configured_credentials:
+                del c["_id"]
+                configured_credentials.append(Credentials.from_mapping(c))
+
+            return configured_credentials
+        except Exception as err:
+            raise RetrievalError(err)
+
+    def get_stolen_credentials(self) -> Sequence[Credentials]:
+        try:
+            stolen_credentials = []
+            list_stolen_credentials = list(mongo.db.stolen_credentials.find({}))
+            for c in list_stolen_credentials:
+                del c["_id"]
+                stolen_credentials.append(Credentials.from_mapping(c))
+
+            return stolen_credentials
+        except Exception as err:
+            raise RetrievalError(err)
```

Comment:
You could reduce duplication by extracting a method that takes a collection as input and returns its contents.